frontend/packages/metal3-plugin/src/components/add-host.tsx

```diff
@@ -0,0 +1,210 @@
+  };
+
+  const setOnlineState = (onlineState) =>
+    setFormValue((prevState) => ({ ...prevState, online: onlineState }));
+
+  const submit = (event) => {
+    event.preventDefault();
+    setProgress(true);
+    setError('');
+    const { name, managementAddress, username, password, online } = formValue;
+    const namespace = getActiveNamespace();
+    const secret = buildSecret(getSecretName(name), namespace, username, password);
+    const bmo = buildBaremetalHostObject(name, namespace, managementAddress, online);
+    createBaremetalHost(secret, bmo)
+      .then(() => {
+        setProgress(false);
+        history.push(resourcePathFromModel(BaremetalHostModel, name, namespace));
+      })
+      .catch(({ message }) => {
+        setProgress(false);
+        setError(message);
+      });
+  };
+
+  return (
+    <React.Fragment>
+      <div className="co-m-pane__body co-m-pane__form">
+        <h1 className="co-m-pane__heading co-m-pane__heading--baseline">
+          <div className="co-m-pane__name">{title}</div>
+        </h1>
+
+        <form onSubmit={submit}>
+          <div className="form-group">
+            <label htmlFor="name" className="control-label co-required">
+              Name
+            </label>
+            <input
+              id="name"
+              name="name"
+              type="text"
+              onChange={handleChange}
+              className="form-control"
+              required
+            />
+          </div>
+
+          <div className="form-group">
+            <label htmlFor="managementAddress" className="control-label co-required">
+              Management Controller Address
+            </label>
+            <input
+              id="managementAddress"
+              name="managementAddress"
+              type="text"
+              onChange={handleChange}
+              className="form-control"
+              required
+            />
+          </div>
+
+          <div className="form-group">
+            <label htmlFor="username" className="control-label co-required">
+              Username
+            </label>
+            <input
+              id="username"
+              name="username"
+              type="text"
+              onChange={handleChange}
+              className="form-control"
+              required
+            />
+          </div>
+
+          <div className="form-group">
+            <label htmlFor="password" className="control-label co-required">
+              Password
+            </label>
+            <input
+              id="password"
+              name="password"
+              type="password"
+              onChange={handleChange}
+              className="form-control"
+              required
+            />
+          </div>
+
+          <div className="formGroup">
+            <p>
+              <Switch
+                id="host-online"
+                label={`Host will be ${formValue.online ? 'online' : 'offline'} after creation.`}
```

Comment:
Making the label dynamic may be slightly confusing, maybe a simple `Make host online after creation` could be better?